filemanager/gcsmanager.go

```diff
@@ -28,18 +30,32 @@ type GCSConfig struct {
 	ForcePathStyle *bool
 	DisableSSL     *bool
 	JSONReads      bool
+
+	uploadIfNotExist bool
+}
+
+type GCSOpt func(*GCSConfig)
+
+func withGCSUploadIfObjectNotExist(uploadIfNotExist bool) func(*GCSConfig) {
```

Comment:
Should we would make it exported since somebody can also create `Manager` directly using `NewGCSManager` which is exported and would need this option to use?
```suggestion
func WithGCSUploadIfObjectNotExist(uploadIfNotExist bool) func(*GCSConfig) {
```